internal/repositories/hash_storage.go

```diff
@@ -0,0 +1,76 @@
+package repositories
+
+import (
+	"database/sql"
+	"github.com/ScienceSoft-Inc/integrity-sum/internal/models"
+	"github.com/ScienceSoft-Inc/integrity-sum/pkg/api"
+	"github.com/sirupsen/logrus"
+)
+
+type HashStorage struct {
+	db     *sql.DB
+	alg    string
+	logger *logrus.Logger
+}
+
+// NewHashStorage creates a new struct HashService
+func NewHashStorage(db *sql.DB, alg string, logger *logrus.Logger) *HashStorage {
+	return &HashStorage{
+		db:     db,
+		alg:    alg,
+		logger: logger,
+	}
+}
+
+// Create accesses the repository to save data to the database
+func (hs HashStorage) Create(allHashData []*api.HashData, deploymentData *models.DeploymentData) error {
+	tx, err := hs.db.Begin()
+	if err != nil {
+		hs.logger.Error("err while saving data in database ", err)
+		return err
+	}
+	query := `INSERT INTO filehashes (full_file_name, hash_sum, algorithm, name_pod, release_id)
+	VALUES($1,$2,$3,$4,(SELECT id from releases WHERE name=$5));`
+
+	for _, hash := range allHashData {
```

Comment:
A bulkInsert function is required instead of one-by-one insert.

Please take a look at `pq` package and `CopyIn()` function.

Reply:
resolved https://github.com/ScienceSoft-Inc/integrity-sum/commit/df701c8e8df4c9c3625d41f08958dcbab2fcd4c6